### DISCUSSION TITLE
FileEdit: Process crashes on file upload

### DISCUSSION THREAD

**Discussion by ckorn** [Megabit/Blazorise#1973] [Q&A]
Hello,

somehow when debugging my process just crashes when I want to upload a file
using the FileEdit field.
When I set a breakpoint in the first line of the OnChanged event, the breakpoint gets hit but then the process
just exists (the console does not show an exception or anything. Just that the process terminated with exit code -1).
My app is a Blazor server app.
In use is Blazorise 0.9.3

This is my code:
```
<FileEdit @ref="fileEdit" Changed="@OnChanged" MaxMessageSize="99999999" />

@code {
    FileEdit fileEdit;
    string sshTarpitLog;

    async Task OnChanged(FileChangedEventArgs e)
    {
        try
        {
            foreach (IFileEntry file in e.Files)
            {
                // A stream is going to be the destination stream we're writing to.
                using (MemoryStream stream = new MemoryStream())
                {
                    // Here we're telling the FileEdit where to write the upload result
                    await file.WriteToStreamAsync(stream);

                    // Once we reach this line it means the file is fully uploaded.
                    // In this case we're going to offset to the beginning of file
                    // so we can read it.
                    stream.Seek(0, SeekOrigin.Begin);

                    // Use the stream reader to read the content of uploaded file,
                    // in this case we can assume it is a textual file.
                    using (StreamReader reader = new StreamReader(stream))
                    {
                        sshTarpitLog = await reader.ReadToEndAsync();
                    }
                }
            }
        }
        finally
        {
            this.StateHasChanged();
        }
    }
}

```

My file is only a few KB in size.
On my deployed machine I also do not get an exception.
The site tells me that it cannot reach the server for a few seconds.
Then it returns but my file is not uploaded.

The entire code can be found here:
https://github.com/ckorn/SshTarpitAnalyzer/blob/master/UI/SshTarpitAnalyzer.Blazor/Pages/SshTarpit.razor

Currently the FileEdit is commented out because I use pastebin to download the log file.

**Comment by stsrki** [comment]
I've copy-pasted the exact same code and it works on my machine. And I also tried your project. It works without any error.

**Reply by ckorn** [reply]
Ok, this is odd.

However in the meantime I found out that it is a different if the site runs in a tab or in a new window.
Looks really strange for me so I made a video:
https://www.youtube.com/watch?v=pISoYeuj7XE

I changed the OnChanged method like this:
```
async Task OnChanged(FileChangedEventArgs e)
    {
        try
        {
            Console.WriteLine("Called OnChanged");
            foreach (IFileEntry file in e.Files)
            {
                Console.WriteLine(file.Name);
                // A stream is going to be the destination stream we're writing to.
                using (MemoryStream stream = new MemoryStream())
                {
                    // Here we're telling the FileEdit where to write the upload result
                    await file.WriteToStreamAsync(stream);
                    Console.WriteLine("After WriteToStreamAsync");

                    // Once we reach this line it means the file is fully uploaded.
                    // In this case we're going to offset to the beginning of file
                    // so we can read it.
                    stream.Seek(0, SeekOrigin.Begin);
                    Console.WriteLine("After Seek");

                    // Use the stream reader to read the content of uploaded file,
                    // in this case we can assume it is a textual file.
                    using (StreamReader reader = new StreamReader(stream))
                    {
                        sshTarpitLog = await reader.ReadToEndAsync();
                        Console.WriteLine($"After ReadToEndAsync: {sshTarpitLog.Length}");
                        Console.WriteLine(sshTarpitLog.Substring(0, 50));
                        this.status = sshTarpitLog.Substring(0, 50);
                    }
                }
            }
        }
        finally
        {
            Console.WriteLine("StateHasChanged");
            this.StateHasChanged();
        }
    }
```

**Comment by ckorn** [comment]
Ok, I found that this workaround also helps here:
https://developercommunity.visualstudio.com/t/1571-iis-express-crash-when-docking-chrome-tab/252050#T-N262921

> I have managed to fix this by unchecking Tools...Options - Projects and Solutions > Web Projects > Stop debugger when browser window is closed. This has reverted to the old behaviour for me.

This fixes the crash here.
Is this an intended fix or just a workaround for a bug?

**Reply by stsrki** [reply]
That is really strange. In my environment "Stop debugger when browser window is closed" is checked and everything works OK.

**Reply by ckorn** [reply]
So should I file a bug report?
Did you see the video?
The crash does not occur when I open a new browser window.

**Reply by stsrki** [reply]
I pulled the latest changes from your project. Followed the video, doing both examples and nothing. No errors. I still can't reproduce the error. Could it be something with different culture on your system?
